### PR TITLE
fix(api-client): harden auth secret extraction with tests

### LINKED ISSUE #8054: @scalar/fastify-api-reference OAuth x-scalar-client-id and clientSecret no longer work
### What happens?

We have this

```
    fastify.register(import("@scalar/fastify-api-reference"), {
      routePrefix: "/",
      configuration: {
        authentication: {
          preferredSecurityScheme: "OAuth2",
          securitySchemes: {
            OAuth2: {
              flows: {
                authorizationCode: {
                  clientSecret: OAUTH_CLIENT_SECRET,
                  "x-scalar-client-id": OAUTH_CLIENT_ID,
                  "x-scalar-security-query": { audience: OAUTH_AUDIENCE },
                },
              },
            },
          },
        },
        hideModels: true,
      },
    })
```

This used to prefill the client ID and client secret but 1.44.11 broke it.

As a side note, @scalar/fastify-api-reference has broken a lot recently, and it's really annoying. Please consider improving your testing, or if this breakage is intentional, make sure you're updating the major version to reflect that.

### What did you expect to happen?

OAuth config to be prefilled

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by amritk**
## Problem

fixes #8054

We had missed a case in this helper, also it was missing tests.

## Solution

Re-wrote it so its fully type safe + tested.

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Medium Risk**
> Touches authentication secret-merging logic used to build request auth, so mistakes could cause missing/incorrect credentials being applied. Risk is mitigated by extensive new unit coverage across scheme types and edge cases.
> 
> **Overview**
> Hardens `extractSecuritySchemeSecrets` to **safely merge secrets** for `apiKey`, `http`, and `oauth2` schemes with explicit precedence (*auth store* > *config* > empty), including better handling of missing/undefined/null OAuth flows and scope selection.
> 
> Adds a large Vitest suite covering scheme types, precedence rules, type mismatches, edge cases, and multi-document/scheme behavior, and updates the API Reference configuration test to use `apiKey.value` (instead of `token`) to match the new extraction behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8eb3b3e45ecc3b4ac594bbaa2860cfbd668aa267. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/api-client/src/v2/blocks/scalar-auth-selector-block/helpers/extract-security-scheme-secrets.ts [hunk 1]
```diff
@@ -21,88 +23,78 @@ import type {
...
+  configSecrets: Record<string, unknown>,
+  authStoreSecrets: Record<string, string> = {},
+): Record<T[number], string> => {
+  const result = {} as Record<T[number], string>
+
+  for (const property of properties) {
+    const authStorevalue = authStoreSecrets[property]
+    const configSecretValue = configSecrets[SECRET_TO_INPUT_FIELD_MAP[property]] as string | undefined
+
+    result[property as T[number]] = authStorevalue || configSecretValue || ''
```
[the remark this answers is not in the document]
Comment (amritk) [reply packages/api-client/src/v2/blocks/scalar-auth-selector-block/helpers/extract-security-scheme-secrets.ts:57]: good bot

[the remark this answers is not in the document]
Comment (amritk) [reply packages/api-client/src/v2/blocks/scalar-auth-selector-block/helpers/extract-security-scheme-secrets.ts:57]: also that was your code
